File: euclidean algorithm/c++/euclidean_algorithm.cpp
#include<iostream>
using namespace std;

int euclidean_GCD_iterative(int a, int b) {
    while(b) {
        int temp = b;
        b = a%b;
        a = temp;
    }
    return a;
}

int euclidean_GCD_recursive(int a, int b) {
    if(b==0)
        return a;
    return euclidean_GCD_recursive(b, a%b); 
}

int main() {
    int a = 8460592, b = 324;
    cout<<euclidean_GCD_iterative(a, b)<<endl;
    cout<<euclidean_GCD_recursive(a, b)<<endl;
}

File: euclidean algorithm/python/euclidean_algorithm.py
def euclidean_GCD_iterative(a, b):
    while b:
        a, b = b, a%b
    return a

def euclidean_GCD_recursive(a, b):
    if b == 0:
        return a
    return euclidean_GCD_recursive(b, a%b)

a = 8460592
b = 324

print(euclidean_GCD_iterative(a, b))
print(euclidean_GCD_recursive(a, b))